### ISSUE TITLE
mmdet写的CrossEntropyLoss与pytorch的CrossEntropyLoss有点不一样

### ISSUE THREAD

**Issue by kache1995** [open-mmlab/mmdetection#9193]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmdetection/issues) and [Discussions](https://github.com/open-mmlab/mmdetection/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (master)](https://github.com/open-mmlab/mmdetection) or [latest version (3.x)](https://github.com/open-mmlab/mmdetection/tree/dev-3.x).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

master branch https://github.com/open-mmlab/mmdetection

### Environment

xxx

### Reproduces the problem - code sample

xxx

### Reproduces the problem - command or script

xxxx

### Reproduces the problem - error message

`   
 # average loss over non-ignored elements
    # pytorch's official cross_entropy average loss over non-ignored elements
    # refer to https://github.com/pytorch/pytorch/blob/56b43f4fec1f76953f15a627694d4bba34588969/torch/nn/functional.py#L2660  # noqa
    if (avg_factor is None) and avg_non_ignore and reduction == 'mean':
        avg_factor = label.numel() - (label == ignore_index).sum().item()
`
mmdet里面的代码写交叉熵损失时，求平均的时候，（有class_weight参数）分母avg_factor简单的使用了样本数量减去忽略类别的样本数量，但是我看pytorch中的交叉熵最后求平均的公式，分母不是样本数量，而是各个样本的class_weight求和。
是故意和pytorch不一样的吗，还是说mmdet这种方式更好

### Additional information

xxx

**Comment by Czm369** [comment]
Probably due to a version issue, none of the algorithms in mmdet seem to assign different weights to different categories.
